src/jit/target.h

```diff
@@ -1384,11 +1384,15 @@ typedef unsigned short regPairNoSmall; // arm: need 12 bits
   #define REG_TAILCALL_ADDR        REG_R1
 
   // The following registers are used in emitting Enter/Leave/Tailcall profiler callbacks
-  #define REG_PROFILER_ENTER_ARG           REG_R0
-  #define RBM_PROFILER_ENTER_ARG           RBM_R0
-  #define REG_PROFILER_RET_SCRATCH         REG_R2
-  #define RBM_PROFILER_RET_SCRATCH         RBM_R2
-  #define RBM_PROFILER_RET_USED            (RBM_R0 | RBM_R1 | RBM_R2)
+  #define REG_PROFILER_ENTER_ARG1           REG_R0
+  #define RBM_PROFILER_ENTER_ARG1           RBM_R0
+  #define REG_PROFILER_ENTER_ARG2           REG_R1
+  #define RBM_PROFILER_ENTER_ARG2           RBM_R1
+  #define REG_PROFILER_RET_SCRATCH1         REG_R2
+  #define RBM_PROFILER_RET_SCRATCH1         RBM_R2
+  #define REG_PROFILER_RET_SCRATCH2         REG_R3
+  #define RBM_PROFILER_RET_SCRATCH2         RBM_R3
+  #define RBM_PROFILER_RET_USED            (RBM_R0 | RBM_R1 | RBM_R2 | RBM_R3)
```

Comment:
The only reason for doing so is that we need to pass the second argument to ProfileLeaveNaked. To preserve this contract, I suppose it would be possible to pass it in r3 instead, and not touch r1/r2? Would this be an acceptable solution?